ca_node/scripts/smach_ranking_controller.py

```diff
@@ -4,40 +4,54 @@
 import smach
 import smach_ros
 
-from ca_msgs.msg import Bumper
+from ca_msgs.msg import Bumper, Cliff
 from geometry_msgs.msg import Twist
 
 is_left_pressed = False
 is_right_pressed = False
+is_cliff_left = False
+is_cliff_right = False
+is_cliff_front_left = False
+is_cliff_front_right = False
+is_cliff = False
+
 
 # define state Forward
 class Forward(smach.State):
     def __init__(self):
-        smach.State.__init__(self, outcomes=['left_pressed', 'right_pressed', 'both_pressed'])
+        smach.State.__init__(self, outcomes=['left_pressed', 'right_pressed', 'both_pressed',
+                                             'cliff_right', 'cliff_left'])
 
     def execute(self, userdata):
         rospy.loginfo('Going forward')
-        global is_left_pressed, is_right_pressed
+        global is_left_pressed, is_right_pressed, is_cliff_left, is_cliff_right, is_cliff_front_left,\
+            is_cliff_front_right, is_cliff
         # Keep in this loop while:
         # LEFT | RIGHT | KEEP?
         #   0  |   0   |  1 --> Keep waiting
         #   0  |   1   |  0 --> right_pressed
         #   1  |   0   |  0 --> left_pressed
         #   1  |   1   |  0 --> both_pressed
-        while not (is_left_pressed or is_right_pressed):
+        while not (is_left_pressed or is_right_pressed or is_cliff):
             global vel_pub
             vel_msg = Twist()
             vel_msg.linear.x = 0.2
             vel_pub.publish(vel_msg)
 
         if is_left_pressed and is_left_pressed:
-          return 'both_pressed'
+            return 'both_pressed'
```

Comment:
Undo these indentation changes (unless they're needed).

Reply:
These changes are in order to follow the PYP8 rules and are marked as errors in the IDE, should I ignore them?

Reply:
No, in that case, keep them.